src/material/schematics/ng-generate/mdc-migration/rules/ts-migration/runtime-migration.ts

```diff
@@ -28,13 +28,54 @@ export class RuntimeCodeMigration extends Migration<ComponentMigrator[], Schemat
       this._migrateModuleSpecifier(node.argument.literal);
     } else if (ts.isImportDeclaration(node)) {
       // Note: TypeScript enforces the `moduleSpecifier` to be a string literal in its syntax.
-      this._migrateModuleSpecifier(node.moduleSpecifier as ts.StringLiteral);
+      this._migrateModuleSpecifier(node.moduleSpecifier as ts.StringLiteral, node.importClause);
+    } else if (this._isNgModuleDecorator(node)) {
+      this._migrateNgModuleImports(node);
     } else if (this._isComponentDecorator(node)) {
-      this._migrateTemplatesAndStyles(node);
+      this._migrateComponentProperties(node);
     }
   }
 
-  private _migrateTemplatesAndStyles(node: ts.Node): void {
+  private _migrateNgModuleImports(node: ts.Node) {
+    // Get the second child node since the first is the decorator symbol
+    node.getChildAt(1).forEachChild(childNode => {
```

Comment:
From my experience, it is untypical to use the non-specific `getChildAt` helpers. Usually what we did was something around this:

```ts
private _migrate(node: ts.Decorator) {
  if (!ts.isCallExpression(node.expression) || node.expression.arguments !== 1) {
    return;
  }

  const call = node.expression;
  const metadata = call.arguments[0]

  if (!ts.isObjectLiteralExpression(metadata) {
    return false;
  }

  for (const prop of metadata.properties) {
    switch (getPropertyNameText(prop)) {
      case "imports": ..
      case "exports": ..
    }
  }
}

export function getPropertyNameText(node: ts.PropertyName): string|null {
  if (ts.isIdentifier(node) || ts.isStringLiteralLike(node)) {
    return node.text;
  }
  return null;
}
```

Reply:
Thanks! I think this made things a lot cleaner so I removed instances of getChildAt